plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -972,34 +981,77 @@ private ContainerInfo createContainerInfo(Service svc,
 
     private List<VolumeMount> getVolumeMountsFrom(String[] volumes, String workspaceID) {
         List<VolumeMount> vms = new ArrayList<>();
-        for (String volume : volumes) {
-            String mountPath = volume.split(":",3)[1];
-            String volumeName = getVolumeName(volume);
-
+        PersistentVolumeClaim pvc = getClaimCheWorkspace();
+        if (pvc != null) {
             VolumeMount vm = new VolumeMountBuilder()
+                .withMountPath("/data")
+                .withName(persistentVolumeClaim)
+                .build();
+            vms.add(vm);
+        } else {
+            for (String volume : volumes) {
+                String mountPath = volume.split(":",3)[1];
+                String volumeName = getVolumeName(volume);
+                VolumeMount vm = new VolumeMountBuilder()
                     .withMountPath(mountPath)
                     .withName("ws-" + workspaceID + "-" + volumeName)
                     .build();
-            vms.add(vm);
+                vms.add(vm);
+            }
         }
         return vms;
     }
 
     private List<Volume> getVolumesFrom(String[] volumes, String workspaceID) {
         List<Volume> vs = new ArrayList<>();
-        for (String volume : volumes) {
-            String hostPath = volume.split(":",3)[0];
-            String volumeName = getVolumeName(volume);
-
+        PersistentVolumeClaim pvc = getClaimCheWorkspace();
+        if (pvc != null) {
+            PersistentVolumeClaimVolumeSource pvcs = new PersistentVolumeClaimVolumeSourceBuilder()
+                  .withClaimName(persistentVolumeClaim)
+                  .build();
             Volume v = new VolumeBuilder()
+                  .withPersistentVolumeClaim(pvcs)
+                  .withName(persistentVolumeClaim)
+                  .build();
+                vs.add(v);
+        } else {
+            for (String volume : volumes) {
+                String hostPath = volume.split(":",3)[0];
+                String volumeName = getVolumeName(volume);
+
+                Volume v = new VolumeBuilder()
                     .withNewHostPath(hostPath)
                     .withName("ws-" + workspaceID + "-" + volumeName)
                     .build();
-            vs.add(v);
+                vs.add(v);
+            }
         }
         return vs;
     }
 
+    private PersistentVolumeClaim getClaimCheWorkspace() {
+        PersistentVolumeClaimList pvcList = openShiftClient.persistentVolumeClaims().list();
+        for(PersistentVolumeClaim pvc:pvcList.getItems()) {
+            if (persistentVolumeClaim.equals(pvc.getMetadata().getName())) {
+                return pvc;
+            }
+        }
+        Map<String, Quantity> requests = new HashMap<>();
+        requests.put("storage", new Quantity("10Gi"));
```

Comment:
Can you extract "10Gi" to a constant. Or better would be if you use a new che property for that to make it configurable.